source/common/quic/quic_transport_socket_factory.h

```diff
@@ -37,13 +37,12 @@ QuicTransportSocketFactoryStats generateStats(Stats::Scope& store, const std::st
 class QuicTransportSocketFactoryBase : public Network::TransportSocketFactory,
                                        protected Logger::Loggable<Logger::Id::quic> {
 public:
-  QuicTransportSocketFactoryBase(Stats::Scope& store, std::unique_ptr<Ssl::ContextConfig> config,
+  QuicTransportSocketFactoryBase(Stats::Scope& store, Ssl::ContextConfig& context_config,
                                  const std::string& perspective)
-      : stats_(generateStats(store, perspective)), context_config_(std::move(config)) {
-    context_config_->setSecretUpdateCallback([this]() {
-      // The callback also triggers updating |context_config_| with the new secret.
-      // TODO(danzh) Client transport socket factory may also need to update quic crypto.
-      stats_.context_config_update_by_sds_.inc();
+      : stats_(generateStats(store, perspective)) {
+    context_config.setSecretUpdateCallback([this]() {
```

Comment:
will this ever be immediately triggered? we'd call a virtual function in the ctor in that case

Reply:
Yes, if triggerred in the callstack, it will call base class version of the virtual function which makes sense as the derived class hasn't use context_config to setup anything yet. Is this not desired in Envoy?

Reply:
https://google.github.io/styleguide/cppguide.html#Doing_Work_in_Constructors

It tends to lead to surprising behavior and is often not resilient to refactors, so I would avoid it if possible